docs/design_docs/milvus_timesync_en.md

```diff
@@ -0,0 +1,114 @@
+# Timesync -- All The things you should know
+
+`Time Synchronization` is the core part of `Milvus 2.0`, it affects all components of the system. This article describes the desgin detail of `Time Synchronization`.
+
+In the `Milvus 2.0`, all events (such as `Create Collection`, `Insert`, `Search`, `Drop Collection`, etc.) have a `Timestamp` to indicate when does this event occurred.
+
+Suppose there are two users, `u1` and `u2`, have connected to the `Milvus`, and do the following actions at their respective timestamp.
+| timestamp | u1                   | u2           |
+|-----------|----------------------|--------------|
+| t0        | create Collection C0 | -            |
+| t2        | -                    | search on C0 |
+| t5        | insert A1 into C0    | -            |
+| t7        | -                    | search on C0 |
+| t10       | insert A2            | -            |
+| t12       | -                    | search on C0 |
+| t15       | delete A1 from C0    | -            |
+| t17       | -                    | search on C0 | 
+
+Ideally, `u2` expects `C0` is empty when it searches at `t2`, and could only sees `A1` at `t7`; at `t12` , the search from `u2` could sees both `A1` and `A2`, but only sees `A2` at `t17`. It's much easier to achieve these targets in `single-node` database. But for `Distributed System` , such like `Milvus`,  it's a little difficult, and the following problems needs to be solved.
+
+1. If `u1` and `u2` are on different nodes, and their time is not synchronized. To give an extreme example, suppose that the time of `u2` is 24 hours later than `u1`, then all the operations of `u1` can't been seen by `u2` until next day.
+2. Network latency. If `u2` starts the `Search on C0` at `t17`, then how to ensure that all the `events` before `t17` have been processed. If the envents of `delete A1 from C0` has been delayed dure to the network latency, then it would lead to incorrect state: `u2` would see both `A1` and `A2` at `t17`.
+
+`Time synchronization system` is used to solve the above problems.
+
+## Timestamp Oracle(TSO)
+
+Like [TiKV](https://github.com/tikv/tikv), `Milvus 2.0` provides `TSO` service, all the events must alloc timestamp from `TSO`，not use local timestamp any more, so the first problem should be solved.
```

Comment:
TSO is a very general solution and it's not invented by tikv.
If we want to talk about timestamp strategy, we may want to mention tidb, cockroach and spanner also

Reply:
File 'internal/tso/tso.go' is copied from 'TiKV', so I mentioned 'TiKV' here